cardano-cli/src/Cardano/CLI/Run/Legacy/Query.hs

```diff
@@ -255,18 +255,20 @@ runQueryProtocolParameters socketPath (AnyConsensusModeParams cModeParams) netwo
     eInMode <- toEraInMode era cMode
       & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
 
-    lift (queryProtocolParameters eInMode sbe)
+    result <- lift (queryProtocolParameters eInMode sbe)
       & onLeft (left . ShelleyQueryCmdUnsupportedNtcVersion)
       & onLeft (left . ShelleyQueryCmdEraMismatch)
 
+    pure (InAnyEra era result)
```

Comment:
Here, it is not possible to return `result` from this computation because it is parameterised by `era` and `era` is not allowed to escape this scope.  By putting `result` in `InAnyEra` it can be smuggled out of this context and used later in `writeProtocolParameters`.

Reply:
See https://github.com/input-output-hk/cardano-cli/pull/162/files#r1290243998